### PR TITLE
docs: add AGENTS.md and CLAUDE.md

### PR DESCRIPTION
**Pull request by Alex-Arsys**
AGENTS.md is the canonical guidance file for AI agents working in this repository — covering architecture constraints, conventions, and rules. CLAUDE.md contains only `@AGENTS.md`, making Claude Code load this file automatically alongside other AI tools.

Assisted-by: ClaudeCode:claude-sonnet-4-6

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+## Rules
+- **Never** modify `SpellChecker/` (pre-compiled binary).
+- **Never** commit `.node` binaries or `pkg` artifacts.
+- **Never** bypass the storage abstraction in `Common/sources/storage/`.
+- **Never** change the co-editing logic in `DocService/sources/` (e.g. `DocsCoServer.js`) without running the full unit test suite (`npm run "unit tests"`).
+- **Always** run `npm run code:check` before PR.
+
+## Findings & Long-tail
+Edge-case debugging recipes and co-editing race conditions belong in the centralized findings store (refer to issue #51).
```
Comment (chrip) [inline AGENTS.md]: I think we need a full URL for `#51`. Couldn't find it.

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+The stateful backend services for Euro-Office Document Server. Forked from `ONLYOFFICE/server`. Licensed **AGPL-3.0**.
+
+## Architecture & Workflow
+- **Async patterns:** Uses `function*` + `co` for async flows. **Do not** refactor to `async/await` unless the entire call chain is updated. Mixing patterns causes race conditions in co-editing state.
+- **Storage:** All I/O must pass through the storage abstraction in `Common/sources/storage/` (`storage-base.js` + per-backend drivers `storage-fs.js` / `storage-s3.js` / `storage-az.js`). Never write directly to disk.
+- **Persistence:** Database schema lives in `schema/<db>/` (one set per backend: `postgresql`, `mysql`, `mssql`, `oracle`, `dameng`), with `createdb.sql` and an `upgrade/` folder. A schema change must be applied across all backends, never mutated ad-hoc.
+- **Packaging:** Uses `pkg`. Adding native modules (`.node`) requires updating `pkg` config.
+
+## Rules
+- **Never** modify `SpellChecker/` (pre-compiled binary).
```
Comment (chrip) [inline AGENTS.md]: It isn't a pre-compiled binary. JS sources are in SpellChecker/sources.

**Review by chrip — CHANGES_REQUESTED** [review]
It just needs a few minor improvements.

File: AGENTS.md (hunk 2 above)
Comment (chrip) [inline AGENTS.md]: `SpellChecker/` contains Node.js source code (`server.js`, `spellCheck.js`, `languages.js`), not a pre-compiled binary. Consider: "Never modify SpellChecker/ without updating its dependencies." or remove the pre-compiled claim.

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+- **Never** commit `.node` binaries or `pkg` artifacts.
+- **Never** bypass the storage abstraction in `Common/sources/storage/`.
+- **Never** change the co-editing logic in `DocService/sources/` (e.g. `DocsCoServer.js`) without running the full unit test suite (`npm run "unit tests"`).
+- **Always** run `npm run code:check` before PR.
+
+## Findings & Long-tail
```
Comment (chrip) [inline AGENTS.md:23]: Issue #51 returns 404 — it doesn't exist in this repository. Either create the issue first, or describe the findings store inline.

**Review by chrip — CHANGES_REQUESTED** [review]
The AGENTS.md is a useful concept and mostly correct. Two factual errors need fixing:

1. **SpellChecker is source code, not a pre-compiled binary.** `SpellChecker/sources/server.js`, `spellCheck.js`, `languages.js` are all Node.js source files. The rule should not call it a "pre-compiled binary."

2. **Issue #51 does not exist.** `GET /issues/51` returns 404. Either create the issue first, or remove the reference.

The rest of the guidance (async patterns, storage abstraction, schema backends, pkg config) is correct and verified against the codebase.

**Comment by chrip** [comment]
Sorry for the duplicate. I posted the first two comments manually. Then, somewhere along the line, a sub-agent finished up and decided on its own to post the same review again. At least we seem to agree :)

File: AGENTS.md (hunk 1 above)
Comment (Alex-Arsys) [reply AGENTS.md]: Done

File: AGENTS.md (hunk 2 above)
Comment (Alex-Arsys) [reply AGENTS.md]: Done

**Review by chrip — APPROVED** [review]
Both issues from my last pass are fixed. Looks good.

**Review by moodyjmz — APPROVED** [review]
Confirmed both prior fixes (SpellChecker source-not-binary, dead #51 reference) and independently re-verified the remaining claims (async co/generator pattern, storage abstraction, schema backends, SpellChecker eslint exclusion, npm scripts, license) against source. LGTM.
